File: bindgen-tests/python/tests/simple_fns.py
from test_package.uniffi_bindgen_tests import *
import unittest

class TestSimpleFns(unittest.TestCase):
    def test_simple_function(self):
        # the test here is just that we can successfully call a function across the FFI
        test_func()

    def test_unexpected_error(self):
        with self.assertRaises(InternalError):
            test_unexpected_error_func()

if __name__ == '__main__':
    unittest.main()

File: bindgen-tests/ruby/tests/simple_fns.rb
# frozen_string_literal: true

# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

require 'test/unit'
require 'uniffi_bindgen_tests'

class TestSimpleFns < Test::Unit::TestCase
  include UniffiBindgenTests

  def test_func
    assert_nothing_raised { UniffiBindgenTests.test_func }
  end

  def test_unexpected_error_func
    # Rust panics are wrapped in UniffiBindgenTests::InternalError
    assert_raise(UniffiBindgenTests::InternalError) do
      UniffiBindgenTests.test_unexpected_error_func
    end
  end
end
